server/routerlicious/packages/services/src/dbFactory.ts

```diff
@@ -0,0 +1,16 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DbFactoryFactory } from "@fluidframework/server-services-core";
+import { Provider } from "nconf";
+import { MongoDbFactory } from "./mongodb";
+export class RouterlicousDbFactoryFactory extends DbFactoryFactory {
```

Comment:
Proposal: instead of adding a FactoryFactory, you could just have a function that returns a Factory.

```ts
interface IDBFactoryConfig {
  name: string;
  modulePathOrName: string;
  dbFactoryConstructorName: string;
  dbConfig: any;
}
export function getDbFactory(config: Provider): IDBFactory {
  const dbFactoryConfig = config.get("db");
  if (!dbFactoryConfig || dbFactoryConfig.name === "mongo") {
    // Taken from current ResourceFactories
    const mongoUrl = config.get("mongo:operationsDbEndpoint") as string;
    const bufferMaxEntries = config.get("mongo:bufferMaxEntries") as number | undefined;
    const mongoFactory = new services.MongoDbFactory(mongoUrl, bufferMaxEntries);
    return mongoFactory;
  }
  const externalDbFactoryModule = require(dbFactoryConfig.modulePathOrName);
  const externalDbFactory = new externalDbFactoryModule[dbFactoryConfig.dbFactoryConstructorName](
    dbFactoryConfig.dbConfig,
  ) as IDBFactory;
  return externalDbFactory;
}
```

I believe you wouldn't need any DBFactoryFactories if you used this or something very similar. And it should maintain back-compat perfectly fine.